### ISSUE TITLE
Về: Kênh cập nhật phiên bản NukeViet

### ISSUE THREAD

**Issue by anhyeuviolet** [nukeviet/nukeviet#2295]
Hiện tại, NukeViet đã có chức năng tự động cập nhật và có thể cập nhật "từ xa" ( OTA hay gì gì đó ). Nhưng vấn đề còn tồn tại đó là, các bản Beta hay Official đều chung 1 "kênh" (Channel).

Điều này phát sinh một tình huống rất dễ xảy ra, đó là người dùng không biết gì cả được giao quản trị 1 trang, thấy cập nhật thích quá bấm cho cập nhật, xong lỗi; dù đó có thể là bản Beta, Develop hay test gì đó.

Do đó mình có ý kiến rằng nên có thêm phần cấu hình, tối thiểu chia thành 2 kênh cập nhật : Phát triển/Ổn định ( Develop/Stable) và cho phép người dùng tuỳ chọn.
Điều này tốt cho người dùng cũng như người quản trị, người phát triển và hỗ trợ có liên quan sẽ tránh được nhiều vấn đề phát sinh trong đáng tiếc.

**Comment by vuthao** [comment]
@hoaquynhtim99 @anhyeuviolet 

Chỉ có phiên bản chính thức mới được cập nhật vào thông báo phiên bản của các site.

Các bản Beta chỉ dành cho nguời dùng hiểu biết sau, hoặc lặp trình viên thử nghiệm phiên bản chính thức. (Thông báo thử nghiệm qua diễn đàn, group, website)

Nếu cái nào chưa đúng thì chúng ta cần sửa lại cho đúng với nguyên tắc này.

**Comment by anhyeuviolet** [comment]
@vuthao Vâng, nếu phân tách ra thì tốt hơn :D Sau này sửa lỗi cho người dùng cuối sẽ khổ lắm. 
Như chính bản 4.1.01 là bản kiểm thử, còn tiềm ẩn lỗi nhưng các bản Chính thức (4.0.29) vẫn được khuyến nghị nâng lên.

**Comment by vuthao** [comment]
@anhyeuviolet Người dùng cuối không có khả năng lập trình thì không nên nâng cấp các bản beta. Bởi quá trình nâng cấp sẽ phực tạp hơn rất nhiều việc cài mới.

**Comment by anhyeuviolet** [comment]
@vuthao dạ thì đúng là thế. Cơ mà hệ thống cứ báo thì họ cứ cho Update, tụi em mới khổ.

**Comment by hoaquynhtim99** [comment]
@vuthao @anhyeuviolet  hiện đã xóa các thông báo cập nhật với bản Beta. Để kho hỗ trợ việc test lỗi hoặc các bản phát triển mình đề nghị sửa như sau:

- Thêm trường phân biệt bản chính thức hay bản thử nghiệm vào các file của ứng dụng (file cài đặt, file nâng cấp)
- Khi thực hiện chức năng kiểm tra phiên bản thì các bản thử nghiệm sẽ không hiển thị
- Khi vào phần quản lý ứng dụng các bản thử nghiệm sẽ hiển thị.
- Nếu bản chính thức tương thích với NukeViet đang chạy thì hiển thị "Tương thích" như hiện tại nếu bản thử nghiệm tương thích thì hiển thị "Thích hợp cho nhà phát triển"
- Khi cài đặt bản dev có thông báo là: Bản này là bản thử nghiệm nếu bạn không am hiểu về hệ thống thì không nên cài đặt.

**Comment by vuthao** [comment]
@hoaquynhtim99 
Theo mình chỉ nên sửa vào 1 chỗ duy nhất là https://nukeviet.vn/vi/store/ 

Với các site: Thêm 1 biến cấu hình vào CSDL: developer mặc định =0. Nếu muốn test các bản thử nghiệm thì cấu hình developer=1
